### ISSUE TITLE
Error when trying to fetch multiple companies

### ISSUE THREAD

**Issue by nixton9** [gadicc/node-yahoo-finance2#445]
# Bug Report

## Describe the bug
I've created a function that receives a ticker and uses 'quote' and 'quoteSummary' to fetch some data for that company. This is working well. Now, I'm trying to create an endpoint that receives an array of tickers and using 'for of' will call my function with each ticker. I would like to use this to get data for all companies on an index (like the S&P 500), so it's a lot of tickers.

What is happening is that everything works fine for the first ~200 companies, but then starts throwing errors for all the other ones, until the last one. I've made tests with different indexes (that have different companies) and the same happens, so it's not related to the actual companies.

Is it normal that this won't work when doing so many requests, one after the other?

Also, thanks for creating and maintaining this library, it's great!

## Minimal Reproduction
```
// My Endpoint

for (const ticker of req.body.tickers) {
      await getStockReport(ticker)
        .then(data => {
         ....
        }
}


// My function
export const getStockReport = async (ticker: string) => {
  const quoteOptions = {
    fields: [
      'longName',
      'symbol',
      'regularMarketPrice',
      'marketCap',
      'epsTrailingTwelveMonths',
      'trailingPE',
      'priceToBook',
      'bookValue',
      'sharesOutstanding',
      'currency'
    ]
  }

  const queryOptions = {
    modules: [
      'financialData',
      'summaryDetail',
      'insiderTransactions',
      'cashflowStatementHistory',
      'balanceSheetHistory',
      'balanceSheetHistoryQuarterly',
      'incomeStatementHistory',
      'earningsTrend'
    ]
  }

  const {
    longName,
    symbol,
    regularMarketPrice,
    marketCap,
    epsTrailingTwelveMonths,
    trailingPE,
    priceToBook,
    bookValue,
    sharesOutstanding,
    currency
  } = await yahooFinance.quote(ticker, quoteOptions as QuoteOptions)

  const {
    financialData: { debtToEquity, currentRatio, returnOnEquity },
    summaryDetail: { dividendYield, dividendRate },
    insiderTransactions,
    balanceSheetHistory: { balanceSheetStatements },
    balanceSheetHistoryQuarterly: { balanceSheetStatements: quarterlyBSS },
    cashflowStatementHistory: { cashflowStatements },
    incomeStatementHistory: { incomeStatementHistory },
    earningsTrend: { trend } = { trend: {} }
  } = Object(
    await yahooFinance.quoteSummary(ticker, queryOptions as QuoteSummaryOptions)
  )

...
```

## Error
It starts throwing this for each company

```
The following result did not validate with schema: #/definitions/QuoteResponseArray
[
  {
    instancePath: '/0',
    schemaPath: '#/definitions/QuoteCryptoCurrency/required',
    keyword: 'required',
    params: { missingProperty: 'circulatingSupply' },
    message: "must have required property 'circulatingSupply'",
    data: {
      language: 'en-US',
      region: 'US',
      quoteType: 'EQUITY',
      triggerable: true,
      quoteSourceName: 'Delayed Quote',
      currency: 'USD',
      exchangeDataDelayedBy: 0,
      regularMarketPrice: 212.25,
      regularMarketTime: 2022-04-01T20:00:02.000Z,
      priceHint: 2,
      fullExchangeName: 'NYSE',
      longName: 'salesforce.com, inc.',
      trailingPE: 116.749176,
      exchange: 'NYQ',
      epsTrailingTwelveMonths: 1.818,
      sharesOutstanding: 985000000,
      bookValue: 57.982,
      marketCap: 209066246144,
      priceToBook: 3.660619,
      market: 'us_market',
      sourceInterval: 15,
      exchangeTimezoneName: 'America/New_York',
      exchangeTimezoneShortName: 'EDT',
      gmtOffSetMilliseconds: -14400000,
      esgPopulated: false,
      tradeable: false,
      marketState: 'CLOSED',
      symbol: 'CRM'
    }
  },
  {
    instancePath: '/0',
    schemaPath: '#/definitions/QuoteCryptoCurrency/required',
    keyword: 'required',
    params: { missingProperty: 'customPriceAlertConfidence' },
    message: "must have required property 'customPriceAlertConfidence'",
    data: {
      language: 'en-US',
      region: 'US',
      quoteType: 'EQUITY',
      triggerable: true,
      quoteSourceName: 'Delayed Quote',
      currency: 'USD',
      exchangeDataDelayedBy: 0,
      regularMarketPrice: 212.25,
      regularMarketTime: 2022-04-01T20:00:02.000Z,
      priceHint: 2,
      fullExchangeName: 'NYSE',
      longName: 'salesforce.com, inc.',
      trailingPE: 116.749176,
      exchange: 'NYQ',
      epsTrailingTwelveMonths: 1.818,
      sharesOutstanding: 985000000,
      bookValue: 57.982,
      marketCap: 209066246144,
      priceToBook: 3.660619,
      market: 'us_market',
      sourceInterval: 15,
      exchangeTimezoneName: 'America/New_York',
      exchangeTimezoneShortName: 'EDT',
      gmtOffSetMilliseconds: -14400000,
      esgPopulated: false,
      tradeable: false,
      marketState: 'CLOSED',
      symbol: 'CRM'
    }
  },
  {
    instancePath: '/0',
    schemaPath: '#/definitions/QuoteCryptoCurrency/required',
    keyword: 'required',
    params: { missingProperty: 'fromCurrency' },
    message: "must have required property 'fromCurrency'",
    data: {
      language: 'en-US',
      region: 'US',
      quoteType: 'EQUITY',
      triggerable: true,
      quoteSourceName: 'Delayed Quote',
      currency: 'USD',
      exchangeDataDelayedBy: 0,
      regularMarketPrice: 212.25,
      regularMarketTime: 2022-04-01T20:00:02.000Z,
      priceHint: 2,
      fullExchangeName: 'NYSE',
      longName: 'salesforce.com, inc.',
      trailingPE: 116.749176,
      exchange: 'NYQ',
      epsTrailingTwelveMonths: 1.818,
      sharesOutstanding: 985000000,
      bookValue: 57.982,
      marketCap: 209066246144,
      priceToBook: 3.660619,
      market: 'us_market',
      sourceInterval: 15,
      exchangeTimezoneName: 'America/New_York',
      exchangeTimezoneShortName: 'EDT',
      gmtOffSetMilliseconds: -14400000,
      esgPopulated: false,
      tradeable: false,
      marketState: 'CLOSED',
      symbol: 'CRM'
    }
  },
  {
    instancePath: '/0',
    schemaPath: '#/definitions/QuoteCryptoCurrency/required',
    keyword: 'required',
    params: { missingProperty: 'lastMarket' },
    message: "must have required property 'lastMarket'",
    data: {
      language: 'en-US',
      region: 'US',
      quoteType: 'EQUITY',
      triggerable: true,
      quoteSourceName: 'Delayed Quote',
      currency: 'USD',
      exchangeDataDelayedBy: 0,
      regularMarketPrice: 212.25,
      regularMarketTime: 2022-04-01T20:00:02.000Z,
      priceHint: 2,
      fullExchangeName: 'NYSE',
      longName: 'salesforce.com, inc.',
      trailingPE: 116.749176,
      exchange: 'NYQ',
      epsTrailingTwelveMonths: 1.818,
      sharesOutstanding: 985000000,
      bookValue: 57.982,
      marketCap: 209066246144,
      priceToBook: 3.660619,
      market: 'us_market',
      sourceInterval: 15,
      exchangeTimezoneName: 'America/New_York',
      exchangeTimezoneShortName: 'EDT',
      gmtOffSetMilliseconds: -14400000,
      esgPopulated: false,
      tradeable: false,
      marketState: 'CLOSED',
      symbol: 'CRM'
    }
  },
  {
    instancePath: '/0',
    schemaPath: '#/definitions/QuoteCryptoCurrency/required',
    keyword: 'required',
    params: { missingProperty: 'toCurrency' },
    message: "must have required property 'toCurrency'",
    data: {
      language: 'en-US',
      region: 'US',
      quoteType: 'EQUITY',
      triggerable: true,
      quoteSourceName: 'Delayed Quote',
      currency: 'USD',
      exchangeDataDelayedBy: 0,
      regularMarketPrice: 212.25,
      regularMarketTime: 2022-04-01T20:00:02.000Z,
      priceHint: 2,
      fullExchangeName: 'NYSE',
      longName: 'salesforce.com, inc.',
      trailingPE: 116.749176,
      exchange: 'NYQ',
      epsTrailingTwelveMonths: 1.818,
      sharesOutstanding: 985000000,
      bookValue: 57.982,
      marketCap: 209066246144,
      priceToBook: 3.660619,
      market: 'us_market',
      sourceInterval: 15,
      exchangeTimezoneName: 'America/New_York',
      exchangeTimezoneShortName: 'EDT',
      gmtOffSetMilliseconds: -14400000,
      esgPopulated: false,
      tradeable: false,
      marketState: 'CLOSED',
      symbol: 'CRM'
    }
  },
  {
    instancePath: '/0/quoteType',
    schemaPath: '#/definitions/QuoteCryptoCurrency/properties/quoteType/const',
    keyword: 'const',
    params: { allowedValue: 'CRYPTOCURRENCY' },
    message: 'must be equal to constant',
    data: 'EQUITY'
  },
  {
    instancePath: '/0',
    schemaPath: '#/definitions/QuoteCurrency/required',
    keyword: 'required',
    params: { missingProperty: 'customPriceAlertConfidence' },
    message: "must have required property 'customPriceAlertConfidence'",
    data: {
      language: 'en-US',
      region: 'US',
      quoteType: 'EQUITY',
      triggerable: true,
      quoteSourceName: 'Delayed Quote',
      currency: 'USD',
      exchangeDataDelayedBy: 0,
      regularMarketPrice: 212.25,
      regularMarketTime: 2022-04-01T20:00:02.000Z,
      priceHint: 2,
      fullExchangeName: 'NYSE',
      longName: 'salesforce.com, inc.',
      trailingPE: 116.749176,
      exchange: 'NYQ',
      epsTrailingTwelveMonths: 1.818,
      sharesOutstanding: 985000000,
      bookValue: 57.982,
      marketCap: 209066246144,
      priceToBook: 3.660619,
      market: 'us_market',
      sourceInterval: 15,
      exchangeTimezoneName: 'America/New_York',
      exchangeTimezoneShortName: 'EDT',
      gmtOffSetMilliseconds: -14400000,
      esgPopulated: false,
      tradeable: false,
      marketState: 'CLOSED',
      symbol: 'CRM'
    }
  },
  {
    instancePath: '/0/quoteType',
    schemaPath: '#/definitions/QuoteCurrency/properties/quoteType/const',
    keyword: 'const',
    params: { allowedValue: 'CURRENCY' },
    message: 'must be equal to constant',
    data: 'EQUITY'
  },
  {
    instancePath: '/0',
    schemaPath: '#/definitions/QuoteEtf/required',
    keyword: 'required',
    params: { missingProperty: 'customPriceAlertConfidence' },
    message: "must have required property 'customPriceAlertConfidence'",
    data: {
      language: 'en-US',
      region: 'US',
      quoteType: 'EQUITY',
      triggerable: true,
      quoteSourceName: 'Delayed Quote',
      currency: 'USD',
      exchangeDataDelayedBy: 0,
      regularMarketPrice: 212.25,
      regularMarketTime: 2022-04-01T20:00:02.000Z,
      priceHint: 2,
      fullExchangeName: 'NYSE',
      longName: 'salesforce.com, inc.',
      trailingPE: 116.749176,
      exchange: 'NYQ',
      epsTrailingTwelveMonths: 1.818,
      sharesOutstanding: 985000000,
      bookValue: 57.982,
      marketCap: 209066246144,
      priceToBook: 3.660619,
      market: 'us_market',
      sourceInterval: 15,
      exchangeTimezoneName: 'America/New_York',
      exchangeTimezoneShortName: 'EDT',
      gmtOffSetMilliseconds: -14400000,
      esgPopulated: false,
      tradeable: false,
      marketState: 'CLOSED',
      symbol: 'CRM'
    }
  },
  {
    instancePath: '/0/quoteType',
    schemaPath: '#/definitions/QuoteEtf/properties/quoteType/const',
    keyword: 'const',
    params: { allowedValue: 'ETF' },
    message: 'must be equal to constant',
    data: 'EQUITY'
  },
  {
    instancePath: '/0',
    schemaPath: '#/definitions/QuoteEquity/required',
    keyword: 'required',
    params: { missingProperty: 'customPriceAlertConfidence' },
    message: "must have required property 'customPriceAlertConfidence'",
    data: {
      language: 'en-US',
      region: 'US',
      quoteType: 'EQUITY',
      triggerable: true,
      quoteSourceName: 'Delayed Quote',
      currency: 'USD',
      exchangeDataDelayedBy: 0,
      regularMarketPrice: 212.25,
      regularMarketTime: 2022-04-01T20:00:02.000Z,
      priceHint: 2,
      fullExchangeName: 'NYSE',
      longName: 'salesforce.com, inc.',
      trailingPE: 116.749176,
      exchange: 'NYQ',
      epsTrailingTwelveMonths: 1.818,
      sharesOutstanding: 985000000,
      bookValue: 57.982,
      marketCap: 209066246144,
      priceToBook: 3.660619,
      market: 'us_market',
      sourceInterval: 15,
      exchangeTimezoneName: 'America/New_York',
      exchangeTimezoneShortName: 'EDT',
      gmtOffSetMilliseconds: -14400000,
      esgPopulated: false,
      tradeable: false,
      marketState: 'CLOSED',
      symbol: 'CRM'
    }
  },
  {
    instancePath: '/0',
    schemaPath: '#/definitions/QuoteIndex/required',
    keyword: 'required',
    params: { missingProperty: 'customPriceAlertConfidence' },
    message: "must have required property 'customPriceAlertConfidence'",
    data: {
      language: 'en-US',
      region: 'US',
      quoteType: 'EQUITY',
      triggerable: true,
      quoteSourceName: 'Delayed Quote',
      currency: 'USD',
      exchangeDataDelayedBy: 0,
      regularMarketPrice: 212.25,
      regularMarketTime: 2022-04-01T20:00:02.000Z,
      priceHint: 2,
      fullExchangeName: 'NYSE',
      longName: 'salesforce.com, inc.',
      trailingPE: 116.749176,
      exchange: 'NYQ',
      epsTrailingTwelveMonths: 1.818,
      sharesOutstanding: 985000000,
      bookValue: 57.982,
      marketCap: 209066246144,
      priceToBook: 3.660619,
      market: 'us_market',
      sourceInterval: 15,
      exchangeTimezoneName: 'America/New_York',
      exchangeTimezoneShortName: 'EDT',
      gmtOffSetMilliseconds: -14400000,
      esgPopulated: false,
      tradeable: false,
      marketState: 'CLOSED',
      symbol: 'CRM'
    }
  },
  {
    instancePath: '/0/quoteType',
    schemaPath: '#/definitions/QuoteIndex/properties/quoteType/const',
    keyword: 'const',
    params: { allowedValue: 'INDEX' },
    message: 'must be equal to constant',
    data: 'EQUITY'
  },
  {
    instancePath: '/0',
    schemaPath: '#/definitions/QuoteMutualfund/required',
    keyword: 'required',
    params: { missingProperty: 'customPriceAlertConfidence' },
    message: "must have required property 'customPriceAlertConfidence'",
    data: {
      language: 'en-US',
      region: 'US',
      quoteType: 'EQUITY',
      triggerable: true,
      quoteSourceName: 'Delayed Quote',
      currency: 'USD',
      exchangeDataDelayedBy: 0,
      regularMarketPrice: 212.25,
      regularMarketTime: 2022-04-01T20:00:02.000Z,
      priceHint: 2,
      fullExchangeName: 'NYSE',
      longName: 'salesforce.com, inc.',
      trailingPE: 116.749176,
      exchange: 'NYQ',
      epsTrailingTwelveMonths: 1.818,
      sharesOutstanding: 985000000,
      bookValue: 57.982,
      marketCap: 209066246144,
      priceToBook: 3.660619,
      market: 'us_market',
      sourceInterval: 15,
      exchangeTimezoneName: 'America/New_York',
      exchangeTimezoneShortName: 'EDT',
      gmtOffSetMilliseconds: -14400000,
      esgPopulated: false,
      tradeable: false,
      marketState: 'CLOSED',
      symbol: 'CRM'
    }
  },
  {
    instancePath: '/0/quoteType',
    schemaPath: '#/definitions/QuoteMutualfund/properties/quoteType/const',
    keyword: 'const',
    params: { allowedValue: 'MUTUALFUND' },
    message: 'must be equal to constant',
    data: 'EQUITY'
  },
  {
    instancePath: '/0',
    schemaPath: '#/definitions/QuoteOption/required',
    keyword: 'required',
    params: { missingProperty: 'customPriceAlertConfidence' },
    message: "must have required property 'customPriceAlertConfidence'",
    data: {
      language: 'en-US',
      region: 'US',
      quoteType: 'EQUITY',
      triggerable: true,
      quoteSourceName: 'Delayed Quote',
      currency: 'USD',
      exchangeDataDelayedBy: 0,
      regularMarketPrice: 212.25,
      regularMarketTime: 2022-04-01T20:00:02.000Z,
      priceHint: 2,
      fullExchangeName: 'NYSE',
      longName: 'salesforce.com, inc.',
      trailingPE: 116.749176,
      exchange: 'NYQ',
      epsTrailingTwelveMonths: 1.818,
      sharesOutstanding: 985000000,
      bookValue: 57.982,
      marketCap: 209066246144,
      priceToBook: 3.660619,
      market: 'us_market',
      sourceInterval: 15,
      exchangeTimezoneName: 'America/New_York',
      exchangeTimezoneShortName: 'EDT',
      gmtOffSetMilliseconds: -14400000,
      esgPopulated: false,
      tradeable: false,
      marketState: 'CLOSED',
      symbol: 'CRM'
    }
  },
  {
    instancePath: '/0',
    schemaPath: '#/definitions/QuoteOption/required',
    keyword: 'required',
    params: { missingProperty: 'expireDate' },
    message: "must have required property 'expireDate'",
    data: {
      language: 'en-US',
      region: 'US',
      quoteType: 'EQUITY',
      triggerable: true,
      quoteSourceName: 'Delayed Quote',
      currency: 'USD',
      exchangeDataDelayedBy: 0,
      regularMarketPrice: 212.25,
      regularMarketTime: 2022-04-01T20:00:02.000Z,
      priceHint: 2,
      fullExchangeName: 'NYSE',
      longName: 'salesforce.com, inc.',
      trailingPE: 116.749176,
      exchange: 'NYQ',
      epsTrailingTwelveMonths: 1.818,
      sharesOutstanding: 985000000,
      bookValue: 57.982,
      marketCap: 209066246144,
      priceToBook: 3.660619,
      market: 'us_market',
      sourceInterval: 15,
      exchangeTimezoneName: 'America/New_York',
      exchangeTimezoneShortName: 'EDT',
      gmtOffSetMilliseconds: -14400000,
      esgPopulated: false,
      tradeable: false,
      marketState: 'CLOSED',
      symbol: 'CRM'
    }
  },
  {
    instancePath: '/0',
    schemaPath: '#/definitions/QuoteOption/required',
    keyword: 'required',
    params: { missingProperty: 'expireIsoDate' },
    message: "must have required property 'expireIsoDate'",
    data: {
      language: 'en-US',
      region: 'US',
      quoteType: 'EQUITY',
      triggerable: true,
      quoteSourceName: 'Delayed Quote',
      currency: 'USD',
      exchangeDataDelayedBy: 0,
      regularMarketPrice: 212.25,
      regularMarketTime: 2022-04-01T20:00:02.000Z,
      priceHint: 2,
      fullExchangeName: 'NYSE',
      longName: 'salesforce.com, inc.',
      trailingPE: 116.749176,
      exchange: 'NYQ',
      epsTrailingTwelveMonths: 1.818,
      sharesOutstanding: 985000000,
      bookValue: 57.982,
      marketCap: 209066246144,
      priceToBook: 3.660619,
      market: 'us_market',
      sourceInterval: 15,
      exchangeTimezoneName: 'America/New_York',
      exchangeTimezoneShortName: 'EDT',
      gmtOffSetMilliseconds: -14400000,
      esgPopulated: false,
      tradeable: false,
      marketState: 'CLOSED',
      symbol: 'CRM'
    }
  },
  {
    instancePath: '/0',
    schemaPath: '#/definitions/QuoteOption/required',
    keyword: 'required',
    params: { missingProperty: 'openInterest' },
    message: "must have required property 'openInterest'",
    data: {
      language: 'en-US',
      region: 'US',
      quoteType: 'EQUITY',
      triggerable: true,
      quoteSourceName: 'Delayed Quote',
      currency: 'USD',
      exchangeDataDelayedBy: 0,
      regularMarketPrice: 212.25,
      regularMarketTime: 2022-04-01T20:00:02.000Z,
      priceHint: 2,
      fullExchangeName: 'NYSE',
      longName: 'salesforce.com, inc.',
      trailingPE: 116.749176,
      exchange: 'NYQ',
      epsTrailingTwelveMonths: 1.818,
      sharesOutstanding: 985000000,
      bookValue: 57.982,
      marketCap: 209066246144,
      priceToBook: 3.660619,
      market: 'us_market',
      sourceInterval: 15,
      exchangeTimezoneName: 'America/New_York',
      exchangeTimezoneShortName: 'EDT',
      gmtOffSetMilliseconds: -14400000,
      esgPopulated: false,
      tradeable: false,
      marketState: 'CLOSED',
      symbol: 'CRM'
    }
  },
  {
    instancePath: '/0',
    schemaPath: '#/definitions/QuoteOption/required',
    keyword: 'required',
    params: { missingProperty: 'strike' },
    message: "must have required property 'strike'",
    data: {
      language: 'en-US',
      region: 'US',
      quoteType: 'EQUITY',
      triggerable: true,
      quoteSourceName: 'Delayed Quote',
      currency: 'USD',
      exchangeDataDelayedBy: 0,
      regularMarketPrice: 212.25,
      regularMarketTime: 2022-04-01T20:00:02.000Z,
      priceHint: 2,
      fullExchangeName: 'NYSE',
      longName: 'salesforce.com, inc.',
      trailingPE: 116.749176,
      exchange: 'NYQ',
      epsTrailingTwelveMonths: 1.818,
      sharesOutstanding: 985000000,
      bookValue: 57.982,
      marketCap: 209066246144,
      priceToBook: 3.660619,
      market: 'us_market',
      sourceInterval: 15,
      exchangeTimezoneName: 'America/New_York',
      exchangeTimezoneShortName: 'EDT',
      gmtOffSetMilliseconds: -14400000,
      esgPopulated: false,
      tradeable: false,
      marketState: 'CLOSED',
      symbol: 'CRM'
    }
  },
  {
    instancePath: '/0',
    schemaPath: '#/definitions/QuoteOption/required',
    keyword: 'required',
    params: { missingProperty: 'underlyingSymbol' },
    message: "must have required property 'underlyingSymbol'",
    data: {
      language: 'en-US',
      region: 'US',
      quoteType: 'EQUITY',
      triggerable: true,
      quoteSourceName: 'Delayed Quote',
      currency: 'USD',
      exchangeDataDelayedBy: 0,
      regularMarketPrice: 212.25,
      regularMarketTime: 2022-04-01T20:00:02.000Z,
      priceHint: 2,
      fullExchangeName: 'NYSE',
      longName: 'salesforce.com, inc.',
      trailingPE: 116.749176,
      exchange: 'NYQ',
      epsTrailingTwelveMonths: 1.818,
      sharesOutstanding: 985000000,
      bookValue: 57.982,
      marketCap: 209066246144,
      priceToBook: 3.660619,
      market: 'us_market',
      sourceInterval: 15,
      exchangeTimezoneName: 'America/New_York',
      exchangeTimezoneShortName: 'EDT',
      gmtOffSetMilliseconds: -14400000,
      esgPopulated: false,
      tradeable: false,
      marketState: 'CLOSED',
      symbol: 'CRM'
    }
  },
  {
    instancePath: '/0/quoteType',
    schemaPath: '#/definitions/QuoteOption/properties/quoteType/const',
    keyword: 'const',
    params: { allowedValue: 'OPTION' },
    message: 'must be equal to constant',
    data: 'EQUITY'
  },
  {
    instancePath: '/0',
    schemaPath: '#/anyOf',
    keyword: 'anyOf',
    params: {},
    message: 'must match a schema in anyOf',
    data: {
      language: 'en-US',
      region: 'US',
      quoteType: 'EQUITY',
      triggerable: true,
      quoteSourceName: 'Delayed Quote',
      currency: 'USD',
      exchangeDataDelayedBy: 0,
      regularMarketPrice: 212.25,
      regularMarketTime: 2022-04-01T20:00:02.000Z,
      priceHint: 2,
      fullExchangeName: 'NYSE',
      longName: 'salesforce.com, inc.',
      trailingPE: 116.749176,
      exchange: 'NYQ',
      epsTrailingTwelveMonths: 1.818,
      sharesOutstanding: 985000000,
      bookValue: 57.982,
      marketCap: 209066246144,
      priceToBook: 3.660619,
      market: 'us_market',
      sourceInterval: 15,
      exchangeTimezoneName: 'America/New_York',
      exchangeTimezoneShortName: 'EDT',
      gmtOffSetMilliseconds: -14400000,
      esgPopulated: false,
      tradeable: false,
      marketState: 'CLOSED',
      symbol: 'CRM'
    }
  }
]

This may happen intermittently and you should catch errors appropriately.
However:  1) if this recently started happening on every request for a symbol
that used to work, Yahoo may have changed their API.  2) If this happens on
every request for a symbol you've never used before, but not for other
symbols, you've found an edge-case.  Please see if anyone has reported
this previously:

  https://github.com/gadicc/node-yahoo-finance2/issues?q=is%3Aissue+Failed%20validation%3A%20%23%2Fdefinitions%2FQuoteResponseArray

or open a new issue (and mention the symbol):

  https://github.com/gadicc/node-yahoo-finance2/issues/new?labels=bug%2C+validation&template=validation.md&title=Failed%20validation%3A%20%23%2Fdefinitions%2FQuoteResponseArray

For information on how to turn off the above logging or skip these errors,
see https://github.com/gadicc/node-yahoo-finance2/tree/devel/docs/validation.md.
```

## Environment
Browser or Node: Node
Node version (if applicable): 14.17.1
Npm version: 6.14.13
Library version (e.g. `1.10.1`): 2.3.0

**Comment by glaucoheitor** [comment]
Hi @nixton9!

Based on my experience, it looks like you're being blocked by Yahoo for sending so many requests.

Have you tried using `quoteCombine` ([doc](https://github.com/gadicc/node-yahoo-finance2/blob/devel/docs/other/quoteCombine.md)) instead of `quote`?

That may improve your experience a little bit, but you might still have issues with `quoteSummary`, maybe you'll want to look on ways on how to spread out your requests instead of sending all at once.

**Comment by gadicc** [comment]
Hey @nixton9; thanks for your detailed error report which is really helpful.

As @glaucoheitor says, it looks like there is some kind of limiting happening on Yahoo's end, but it's not their usual kind of block (which we avoid with our own concurrency limits).  We're still getting (most of) the result, in this case, missing just the `customPriceAlertConfidence` property which is usually always present.  I have no idea what logic they employ to handle such cases ("if many requests, drop expensive fields" I guess, but is this the only field?).

I'm going to suggest that you edit `node_modules/yahoo-finance2/dist/esm/schema.json.js` (or `cjs` if you're using `require`), search for the `QuoteEquity` object, go down to the `required` sub-property and comment out the line `"customPriceAlertConfidence"` and let us know know if that works around this.

Since I can't test this easily, I just want to see if that is the only field that they suddenly stop providing or if there are other changes too.  But reading through all the errors, it doesn't look like any fields are missing (at least not any other `required` fields).

**Comment by gadicc** [comment]
And definitely, as @glaucoheitor says, try use `quoteCombine` for all the `quote` queries.  Although, I have a feeling you might run into a limit on the max number of queries you can send at once... so please report back on that ;)  And then, try do the more expensive `quoteSummary` later on.  You could also try adding a delay to all your requests.  Assuming you're storing this all in a local database; how often do you intend to refresh both the quotes and quoteSummaries?

Thanks also for your kind words about the lib :)

**Comment by nixton9** [comment]
Thanks @glaucoheitor  and @gadicc .
Will try this out when I have some time and will report my findings here.
